### PR TITLE
User permissions for setting up Warehouse Actions sources

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: warehouse-actions/google-bigquery.md [hunk 1]
```diff
@@ -4,40 +4,75 @@ description: Step-by-step guide to ingest your data from Google BigQuery into Ru
...
+![](https://user-images.githubusercontent.com/59817155/133751255-356dab76-a795-4428-8e72-9c46b0031d79.png)
+
+* In the resulting pop-up, select **JSON** and click on **CREATE**.
+
+![](https://user-images.githubusercontent.com/59817155/133751286-a7897da9-eb9d-48ef-be29-f16f0e65e2bb.png)
+ 
+* Finally, download this JSON file. This file is required while creating a BigQuery warehouse source in RudderStack - the next section covers the steps to do this.
+
+{% hint style="info" %}
+RudderStack creates a dedicated dataset called `rudderstack` in your project that is used to store the state of each data sync.
```
Comment (ashishRudder) [inline warehouse-actions/google-bigquery.md:72]: `rudderstack` needs to be updated with `rudderstack_`

**Review by ashishRudder — CHANGES_REQUESTED** [review]

File: warehouse-actions/amazon-redshift.md [hunk 2]
```diff
@@ -10,16 +10,76 @@ This guide will help you configure Redshift as a source from which you can route
...
+
+``` 
+ALTER DEFAULT PRIVILEGES IN SCHEMA "<YOUR_SCHEMA>" GRANT SELECT ON TABLES TO rudder;
+```
+
+This command allows the user `rudder` to read the data from all the future tables in the schema `<YOUR_SCHEMA>`.
+
 
-## Set Up as Source
+## Setting Up the Source
```
[the remark this answers is not in the document]
Comment (RanjeetMishra) [reply warehouse-actions/amazon-redshift.md:82]: This comment applies for all WH sources.

[the remark this answers is not in the document]
Comment (ameypv-rudder) [reply warehouse-actions/amazon-redshift.md:82]: sure - will look into this for all the WH Actions sources.

File: warehouse-actions/amazon-redshift.md [hunk 3]
```diff
@@ -10,16 +10,76 @@ This guide will help you configure Redshift as a source from which you can route
...
+ 
+This command allows the user `rudder` to have full access to all the objects that exist in the schema `_rudderstack`.
+
+
+**Optional**: Run the following commands to allow the user `rudder` to read the data from all the tables in the schema <YOUR_SCHEMA>:
+
+``` 
+GRANT SELECT ON ALL TABLES IN SCHEMA "<YOUR_SCHEMA>" TO rudder;
+```
+This command allows the user `rudder` to read data from all the tables in the schema `<YOUR_SCHEMA>`.
```
[the remark this answers is not in the document]
Comment (ameypv-rudder) [reply warehouse-actions/amazon-redshift.md]: Slightly tweaked the sentence to make the idea clearer.

File: warehouse-actions/amazon-redshift.md [hunk 4]
```diff
@@ -10,16 +10,84 @@ This guide will help you configure Redshift as a source from which you can route
...
+
+```
+GRANT SELECT ON TABLE "<YOUR_SCHEMA>"."<YOUR_TABLE>" TO rudder;
 ```
 
-## Set Up as Source
+This command allows the user `rudder` to read the data from specified table `<YOUR_TABLE>`. Replace `<YOUR_SCHEMA>` and `<YOUR_TABLE>` with the exact names of your Redshift schema and table.
+
+
+### Optional permissions to grant on the schema objects
```
Comment (ashishRudder) [inline warehouse-actions/amazon-redshift.md:74]: it should be written as schemas' objects?
We are providing grant on objects inside schema.

File: warehouse-actions/postgresql.md [hunk 5]
```diff
@@ -8,18 +8,73 @@ PostgreSQL is an enterprise-grade, open source database management system. It su
...
+
+This command lets the user `rudder` look up objects within the schema `<YOUR_SCHEMA>`. Replace <YOUR_SCHEMA> with the exact name of your PostgreSQL database schema.
+
+``` 
+GRANT SELECT ON TABLE "<YOUR_SCHEMA>"."<YOUR_TABLE>" TO rudder;
+```
+ 
+This command allows the user `rudder` to read data from the table `<YOUR_TABLE>`. Replace <YOUR_SCHEMA> and <YOUR_TABLE> with the exact name of your database schema and table names in PostgreSQL.
+
+### Optional permissions to grant on the schema objects
```
Comment (ashishRudder) [inline warehouse-actions/postgresql.md:60]: same here.

**Review by RanjeetMishra — COMMENTED** [review]
LGTM except BigQuery.
Fort BigQuery we are not mentioning user to create rudderstack_ schema

**Comment by ameypv-rudder** [comment]
> LGTM except BigQuery.
> Fort BigQuery we are not mentioning user to create rudderstack_ schema

Hey Ranjeet, as discussed this information is already present and will be visible to the user as a tip/info box.

File: warehouse-actions/amazon-redshift.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ameypv-rudder) [reply warehouse-actions/amazon-redshift.md:94]: I have the FAQ from @ashishRudder on this - will update all the WH actions docs with the screenshot and the info.

**Review by RanjeetMishra — APPROVED** [review]

**Comment by ashishRudder** [comment]
> LGTM except BigQuery.
> Fort BigQuery we are not mentioning user to create rudderstack_ schema

It has been added in the latest commits.

**Review by ashishRudder — APPROVED** [review]

**Pull request merged**